mt_gateway_test.go

```diff
@@ -0,0 +1,369 @@
+package main
```

Comment:
I wish I could :slightly_smiling_face: 
That's how I initially wanted to have these tests but `main` is not a package, it's a program thus it is not importable.

I briefly explained this in an earlier PR of mine where I had to remove `TestMain` from `docker_test.go` (see [this PR](https://github.com/rudderlabs/rudder-server/pull/2059)) in order to have this test coexist in the same folder.

I'm doing it this way to avoid building an image of RudderServer and have to spawn yet another container, which is the only other solution that I can think of. The alternative is what I'm doing here i.e.:

```go
	done := make(chan struct{})
	go func() {
		defer close(done)
		Run(ctx)
	}()
```

:point_up: much cheaper than building a Docker image and running it as a container.

If you know of another way of course I'm all :ear: :ear:

Reply:
Keeping only `func main()` and `version` vars in `main` and moving everything else to another package would work?

If it is risky or too much work don't bother